boot/kernel_os.go

```diff
@@ -36,20 +36,25 @@ var _ BootParticipant = (*coreBootParticipant)(nil)
 
 func (*coreBootParticipant) IsTrivial() bool { return false }
 
-func (bp *coreBootParticipant) SetNextBoot() (rebootRequired bool, err error) {
+func (bp *coreBootParticipant) SetNextBoot() (rebootInfo RebootInfo, err error) {
 	const errPrefix = "cannot set next boot: %s"
 
 	rebootRequired, u, err := bp.bs.setNext(bp.s)
 	if err != nil {
-		return false, fmt.Errorf(errPrefix, err)
+		return RebootInfo{RebootRequired: false}, fmt.Errorf(errPrefix, err)
 	}
 
 	if u != nil {
 		if err := u.commit(); err != nil {
-			return false, fmt.Errorf(errPrefix, err)
+			return RebootInfo{RebootRequired: false}, fmt.Errorf(errPrefix, err)
 		}
 	}
-	return rebootRequired, nil
+
+	rbl, err := bp.bs.getRebootBootloader()
```

Comment:
should we do this if rebootRequire is false?

Reply:
The interface should not be used in that case, but yeah, it is probably safer to make it nil when no reboot is required. Changed.

Reply:
This does not apply anymore as now setNext() returns the full RebootInfo object.